Realm/RLMGeospatial.h

```diff
@@ -0,0 +1,248 @@
+ and 0 or more inner `Polygon`s, called `holes`, which represents an unlimited number of internal holes
+ inside the outer `Polygon`.
+ A `Polygon` describes a shape conformed by at least three segments, where the last and the first `RLMGeospatialPoint`
+ must be the same to indicate a closed polygon (meaning you need at least 4 points to define a polygon).
+ Inner holes in a `RLMGeospatialPolygon` must  be entirely inside the outer ring
+
+ A `hole` has the following restrictions:
+ - Holes may not cross, i.e. the boundary of a hole 
+ may not intersect both the interior and the exterior of any other
+   hole.
+ - Holes may not share edges, i.e. if a hole contains and edge AB, the no other hole may contain it.
+ - Holes may share vertices, however no vertex may appear twice in a single hole.
+ - No hole may be empty.
+ - Only one nesting is allowed.
+
+ @warning This class cannot be persisted and can only be use within a geospatial `geoWithin` query.
+ */
+@interface RLMGeospatialPolygon : NSObject <RLMGeospatial>
+/// The polygon's external (outer) ring.
+@property (readonly, strong) NSArray<RLMGeospatialPoint *> *outerRing;
+/// The holes (if any) in the polygon.
+@property (readonly, strong, nullable) NSArray<NSArray<RLMGeospatialPoint *> *> *holes;
+
+/**
+Initialize a `RLMGeospatialPolygon`, with its outer rings and holes (if any).
+
+Returns `nil` if the `RLMGeospatialPoints` representing a polygon (outer ring or holes), don't have at least 4 points.
+Returns `nil` if the first and the last `RLMGeospatialPoint` in a polygon are not the same.
+
+@param outerRing The polygon's external (outer) ring.
+ */
+- (nullable instancetype)initWithOuterRing:(NSArray<RLMGeospatialPoint *> *)outerRing;
+
+/**
+Initialize a `RLMGeospatialPolygon`, with its outer rings and holes (if any).
+
+Returns `nil` if the `RLMGeospatialPoints` representing a polygon (outer ring or holes), don't have at least 4 points.
+Returns `nil` if the first and the last `RLMGeospatialPoint` in a polygon are not the same.
+
+@param outerRing The polygon's external (outer) ring.
+@param holes The holes (if any) in the polygon.
+ */
+- (nullable instancetype)initWithOuterRing:(NSArray<RLMGeospatialPoint *> *)outerRing holes:(nullable NSArray<NSArray<RLMGeospatialPoint *> *> *)holes;
+@end
+
+/**
+ This structure is a helper to represent/convert a distance. It can be used in geospatial
+ queries like those represented by a `RLMGeospatialCircle`
+
+ - warning: This structure cannot be persisted and can only be used to build other geospatial shapes
+ */
+@interface RLMDistance : NSObject
+/// The distance in radians.
+@property (readonly) double radians;
+
+/**
+Constructs a `Distance`.
+
+Returns `nil` if the value is lower than 0, because we cannot construct negative distances.
+@param kilometers Distance in kilometers.
+@returns A value that represents the provided distance in radians.
+ */
++ (nullable instancetype)kilometers:(double)kilometers;
```

Comment:
`distanceFromKilometers:(double)kilometers NS_SWIFT_NAME(kilometers:)` and similar for each of these.